x/auth/tx/service_test.go

```diff
@@ -139,14 +156,53 @@ func (s IntegrationTestSuite) TestGetTxEvents() {
 	s.Require().Equal(len(grpcRes.Txs), 1)
 	s.Require().Equal("foobar", grpcRes.Txs[0].Body.Memo)
 
-	// Query the tx via grpc-gateway.
-	restRes, err := rest.GetRequest(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?event=%s&pagination.offset=%d&pagination.limit=%d", val.APIAddress, "message.action=send", 0, 1))
-	s.Require().NoError(err)
-	var getTxRes tx.GetTxsEventResponse
-	s.Require().NoError(val.ClientCtx.JSONMarshaler.UnmarshalJSON(restRes, &getTxRes))
-	s.Require().Equal(len(grpcRes.Txs), 1)
-	s.Require().Equal("foobar", getTxRes.Txs[0].Body.Memo)
-	s.Require().NotZero(grpcRes.TxResponses[0].Height)
+	rpcTests := []struct {
+		name      string
+		url       string
+		expectErr bool
+		expected  *tx.GetTxsEventResponse
+	}{
+		{
+			"empty params",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs", val.APIAddress),
+			true,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"without pagination",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?event=%s", val.APIAddress, "message.action=send"),
+			false,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"with pagination",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?event=%s&pagination.offset=%d&pagination.limit=%d", val.APIAddress, "message.action=send", 0, 10),
+			false,
+			&tx.GetTxsEventResponse{},
+		},
+		{
+			"expect pass with multiple-events",
+			fmt.Sprintf("%s/cosmos/tx/v1beta1/txs?event=%s&event=%s", val.APIAddress, "message.action=send", "message.module=bank"),
```

Comment:
OK, so it works without escaping.

Could you add a test to test if it works with escaping too?